lib/ui/painting/display_list_image_gpu.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_DISPLAY_LIST_IMAGE_GPU_H_
+#define FLUTTER_LIB_UI_PAINTING_DISPLAY_LIST_IMAGE_GPU_H_
+
+#include "flutter/display_list/display_list_image.h"
+#include "flutter/flow/skia_gpu_object.h"
+#include "flutter/fml/macros.h"
+
+namespace flutter {
+
+class DlImageGPU final : public DlImage {
+ public:
+  static sk_sp<DlImageGPU> Make(SkiaGPUObject<SkImage> image);
+
+  // |DlImage|
+  ~DlImageGPU() override;
+
+  // |DlImage|
+  sk_sp<SkImage> skia_image() const override;
+
+  // |DlImage|
+  std::shared_ptr<impeller::Texture> impeller_texture() const override;
+
+  // |DlImage|
+  bool isTextureBacked() const override;
+
+  // |DlImage|
+  SkISize dimensions() const override;
+
+  // |DlImage|
+  virtual size_t GetApproximateByteSize() const override;
+
+ private:
+  SkiaGPUObject<SkImage> image_;
```

Comment:
Why is this held here in a SkiaGPUObject rather than in the ./image.h object using that wrapper? Wrapping it here in this object means that anyone who uses this needs to go through the unref thread handoff when the image objects should be direct unref-able by the engine without that overhead.

Reply:
We discussed this over VC and the consensus was that dropping the DlImageGPU reference would not unref SkiaGPUObject object. So the refcount churn of that object wouldn't be significantly different.

Reply:
Resolved in chat. Since the field is only deref'd when the DlImageGPU reaches end of life - which happens when the "image" itself reaches end of life, there will only ever be on case of enqueuing an deref of the wrapped image here.